chain/client/src/test_utils/test_env.rs

```diff
@@ -58,6 +57,12 @@ pub struct TestEnv {
     pub(crate) save_trie_changes: bool,
 }
 
+pub struct StateWitnessPropagationResult {
+    /// Whether some propagated state witness includes two different post state
+    /// roots.
+    pub found_different_post_state_roots: bool,
```

Comment:
May I suggest this to be renamed to explicitly mention that this comes from implicit transitions? Something like, found_differing_post_state_root_due_to_implicit_transitions

because at first read, I thought this was an error condition - like, maybe we're supposed to have the same state roots but somehow we found different ones.

Reply:
Sure! To be clear, these are all state transitions, not just implicit.